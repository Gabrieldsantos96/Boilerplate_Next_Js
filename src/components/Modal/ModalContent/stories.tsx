import { Meta, StoryObj } from '@storybook/react'
import ModalContent from '.'

export default {
  title: 'ModalContent',
  component: ModalContent
} as Meta

export const Default: StoryObj = {}
