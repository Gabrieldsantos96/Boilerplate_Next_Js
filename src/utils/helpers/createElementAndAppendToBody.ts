export function createElementAndAppendToBody(containerId: string) {
  const wrapperElement = document.createElement('div')
  wrapperElement.setAttribute('id', containerId)
  wrapperElement.setAttribute('data-testid', containerId)
  document.body.appendChild(wrapperElement)
  return wrapperElement
}
