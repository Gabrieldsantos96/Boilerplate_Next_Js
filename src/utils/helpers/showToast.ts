import { EventManagerDataTransfer } from '@/components/Toast/ToastContainer/useToastContainer'
import EventManager from '@/lib/eventManager'

export const toastEventManager = new EventManager<EventManagerDataTransfer>()

export const showToast = ({
  type,
  text,
  duration
}: EventManagerDataTransfer) => {
  toastEventManager.emit('addToast', { type, text, duration })
}
