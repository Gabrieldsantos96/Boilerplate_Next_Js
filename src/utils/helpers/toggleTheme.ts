import EventManager from '@/lib/eventManager'

export const themeEventManager = new EventManager<string>()

export const toggleTheme = () => {
  themeEventManager.emit('toggleTheme', '')
}
