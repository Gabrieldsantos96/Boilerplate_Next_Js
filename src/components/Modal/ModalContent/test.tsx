import { render } from '@/utils/helpers/testUtils'
import ModalContent from '.'

describe('<ModalContent />', () => {
  it('should render the content', () => {
    const { container } = render(<ModalContent>Test Content</ModalContent>)
    expect(container.textContent).toContain('Test Content')
    expect(container.firstChild).toMatchSnapshot()
  })
})
