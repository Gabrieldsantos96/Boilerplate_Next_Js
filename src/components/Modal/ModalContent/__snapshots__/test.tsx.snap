// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalContent /> should render the content 1`] = `
.c0 {
  margin-top: 3.2rem;
}

<main
  class="c0"
>
  Test Content
</main>
`;
